source/Ship.h

```diff
@@ -203,8 +203,8 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	bool CanSendHail(const PlayerInfo &player, bool allowUntranslated = false) const;
 
 	// Access the ship's AI cache, containing the range and expected AI behavior for this ship.
-	ShipAICache &GetAICache();
-	void UpdateCaches();
+	const ShipAICache &GetAICache() const;
+	void UpdateCaches(bool massLessChange = false);
```

Comment:
Can you add a comment explaining what the parameter means?

Reply:
Sorry, it's been almost a year since I worked on this, and at this point, I'm basically trying to figure out what everything does again. So a comment is not going to be immediately forthcoming.

Reply:
```suggestion
	// Updates the AI and navigation caches. If the ship's mass hasn't changed,
	// reuses some of the previous values.
	void UpdateCaches(bool massLessChange = false);
```
Maybe something like this would work?